include/fbgemm/FbgemmBuild.h

```diff
@@ -25,14 +26,19 @@
         #define FBGEMM_API __declspec(dllimport)
       #endif
     #endif
+    #define FBGEMM_ENUM_CLASS_API
   #else
-    // "GNUC with version < 6 have a bug if visibility attribute is used on an enum class.
-    // https://stackoverflow.com/questions/2463113/g-c0x-enum-class-compiler-warnings "
-    #if __clang__ || __GNUC__ >=6 || __INTEL_COMPILER
+    #if __clang__ || __GNUC__ >= 4 || __INTEL_COMPILER
       #define FBGEMM_API __attribute__((__visibility__("default")))
     #else
       #define FBGEMM_API
     #endif
+    // Currently, enum classes need to be declaredly explicitly for shared build on macos
+    #if __clang__
```

Comment:
clang builds are used on linux as well.

Reply:
Need to find whether this is related with __clang__ or __APPLE__.